### DISCUSSION TITLE
useGLBF and MongoDB

### DISCUSSION THREAD

**Discussion by Eliott-Sarrey** [pmndrs/drei#598] [Q&A]
Hello, I would like to render a GLB file stored in my MongoDB database using DREI.
What I fetch is a string with the content of the GLB. In my previous tests, I was using the function useGLBF and a local file. However, this function does not accept any string. Does anyone know how I could parse my file (using loader.parse() I guess) in order to be able to use this function and DREI?

Thank you in advance!
Best regards, 
Eliott

**Comment by ashleymvanduzer** [comment]
https://threejs.org/docs/#examples/en/loaders/GLTFLoader maybe? and run an .onload to wait for it to load completely before rendering? sorry, this answer might be too directly threejs oriented. I would consider using gltfjsx --transform, if can significantly reduce the size of any gltf model.

**Comment by drcmda** [comment]
if you want the same caching behaviour and suspense integration:

```jsx
import suspend from 'suspend-react'
import { GLTFLoader } from 'three-stdlib'

function Model() {
  const { scene } = suspend(() => new Promise((res, rej) => new GLTFLoader().parse(buffer, "/", res, rej), [buffer])
```
